lib/matroid/detector.rb

```diff
@@ -0,0 +1,314 @@
+# error messages
+DETECTOR_DOES_NOT_EXITS_ERR = "Error: Couldn't find detector"
+DETECTOR_NOT_TRAINED_ERR = "Error: This detector has incomplete training."
+
+# size limit constants
+IMAGE_FILE_SIZE_LIMIT = 50 * 1024 * 1024
+VIDEO_FILE_SIZE_LIMIT = 300 * 1024 * 1024
+BATCH_FILE_SIZE_LIMIT = 50 * 1024 * 1024
+ZIP_FILE_SIZE_LIMIT   = 300 * 1024 * 1024
+module Matroid
+
+
+  # Represents a matroid detector
+  # @attr [String]          id                Detector id
+  # @attr [String]          name              Detector name
+  # @attr [Array<String>]   labels
+  # @attr [String]          permission_level  'private', 'readonly', 'open', 'stock'
+  # @attr [Bool]            owner             is the current authicated user the owner
+  class Detector
+    # HASH { <id> => Detector }
+    @@detectors = {}
```

Comment:
This variable isn't available outside. Think of this class as something like an ORM. The reason for storing the detectors as a hash on the class was to have methods like 
```
Matroid::Detector.find(name: /cat/, state: 'trained')
```
which would search the detectors in storage.